girder/utility/gridfs_assetstore_adapter.py

```diff
@@ -79,31 +84,45 @@ def validateInfo(doc):
 
     @staticmethod
     def fileIndexFields():
-        return ['sha512']
+        return ['sha512', 'chunkUuid']
 
     def __init__(self, assetstore):
         """
         :param assetstore: The assetstore to act on.
         """
         super(GridFsAssetstoreAdapter, self).__init__(assetstore)
+        recent = False
+        connectionArgs = (self.assetstore.get('mongohost', None),
+                          self.assetstore.get('replicaset', None))
+        try:
+            # Guard in case the connectionArgs is unhashable
+            key = connectionArgs
+            if key in _recentConnections:
+                recent = (time.time() - _recentConnections[key]['created'] <
+                          RECENT_CONNECTION_CACHE_TIME)
+        except Exception:
+            key = None
         try:
-            self.chunkColl = getDbConnection(
-                self.assetstore.get('mongohost', None),
-                self.assetstore.get('replicaset', None)
-            )[self.assetstore['db']].chunk
-            _ensureChunkIndices(self.chunkColl)
+            # MongoClient automatically reuses connections from a pool, however
+            # getting the connection takes time, so use our own cache.  We also
+            # avoid redoing ensureChunkIndices, which is also takes time.
+            self.chunkColl = getDbConnection(*connectionArgs)[self.assetstore['db']].chunk
+            if not recent:
+                _ensureChunkIndices(self.chunkColl)
+                if key is not None:
+                    _recentConnections[key] = {
+                        'created': time.time()
+                    }
```

Comment:
Yes.  I could add a simplistic purge when it gets to a certain size.  Since the individual savings is small, just clearing the cache when it hits a size threshold would probably be sufficient (something like `if len(_recentConnections) > MAX_CACHE_SIZE: _recentConnections.clear()`).